### PR TITLE
Add github token when cloning cadl repo

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Set GitHub PAT as env variable to clone private repo

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
One question related to multiple reviews but otherwise looks good assuming the pat has the necessary permissions to the repo.

File: eng/scripts/Create-Apiview-Token-Cadl.ps1 [hunk 1]
```diff
@@ -78,7 +79,13 @@ if ($revs)
...
         if ($GitRepoName -ne $prevRepo)
         {
-            git clone --no-checkout --filter=tree:0 "https://github.com/$GitRepoName"
+            $gitUrl = "https://github.com"
+            if ($GitPat)
+            {
+                $gitUrl = "https://$GitPat@github.com"
+            }
+
+            git clone --no-checkout --filter=tree:0 "$gitUrl/$GitRepoName"
```
Comment (weshaggard) [inline eng/scripts/Create-Apiview-Token-Cadl.ps1:88]: Not introduced with this change but this multiple clone in the same directory might cause issues with multiple reviews.

**Comment by praveenkuttappan** [comment]
Fixes https://github.com/Azure/azure-sdk-tools/issues/4975

**Pull request merged**